examples/zephyr/golioth_basics/Kconfig

```diff
@@ -3,12 +3,12 @@
 
 mainmenu "Golioth application options"
 
-config GOLIOTH_PSK_ID
+config GOLIOTH_SAMPLE_PSK_ID
```

Comment:
This is duplicated in `examples/zephyr/common/Kconfig` now, right?

Reply:
My intent was to leave golioth_basics as a more verbose example. So the client config is not handled by the common helper but shown in the app_main.c using these symbols. The prj.conf contains the symbols that would otherwise be set in the `common/Kconfig.defconfig` file.

Reply:
I see. It's confusing to have 2 symbols with the same name defined in different places, even if they aren't included in the same build ever. Can we change the name to make them distinct?

Reply:
I made the following changes to golioth_basic regarding this conversation:
* Removed Kconfig symbols from the project in favor of those in the common library
* Added `CONFIG_GOLIOTH_SAMPLE_HARDCODED_CREDENTIALS=y` so those symbols are available
* Left the client config struct in the example as that is the bit that I think is important for users to understand from this example.